src/oracle/connection/stmt.rs

```diff
@@ -135,8 +180,16 @@ impl Statement {
                 ptr::null_mut(),
                 ffi::OCI_DEFAULT,
             );
-            Self::check_error(self.connection.env.error_handle, status)?;
+            Self::check_error_sql(
+                self.connection.env.error_handle,
+                status,
+                &self.mysql,
+                "EXECUTING STMT".to_string(),
+            )?;
         }
+        // TODO: since we have a statement cache, which I couldn't find to turn off,
+        // we need to reset the bind_index once we executed the statement
+        self.bind_index = 0;
```

Comment:
Why is this needed? I do not see where bind_index is set to a not 0 value?

Reply:
This happens when the statement is cached, which occurs if you try to execute the same statement with varying parameters. Since the binding is re-executed the internal counter will continue counting where it stopped before. Setting this to 0 after a statement has been run will start with 1 if new parameters are used with an existing statement.

Reply:
The comment should say something like this:
"The bind index is required to start by zero. If a statement is executed more than once we need to reset the index here"

(The comment should not mention something about the statement cache, because we want this cache :wink:)